yearn/hardhat.config.ts

```diff
@@ -0,0 +1,76 @@
+import { HardhatUserConfig } from "hardhat/config"
+
+import "@keep-network/hardhat-helpers"
+import "@nomiclabs/hardhat-waffle"
+import "hardhat-gas-reporter"
+import "hardhat-contract-sizer"
+import "hardhat-deploy"
+
+const config: HardhatUserConfig = {
+  solidity: {
+    compilers: [
+      {
+        version: "0.6.12",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+    ],
+  },
+
+  paths: {
+    artifacts: "./build",
+  },
+
+  networks: {
+    hardhat: {
+      forking: {
+        // forking is enabled only if FORKING_URL env is provided
+        enabled: !!process.env.FORKING_URL,
+        // URL should point to a node with archival data (Alchemy recommended)
+        url: process.env.FORKING_URL || "",
+        // latest block is taken if FORKING_BLOCK env is not provided
+        blockNumber:
+          process.env.FORKING_BLOCK && parseInt(process.env.FORKING_BLOCK),
+      },
+      tags: ["local"]
+    },
+    development: {
+      url: "http://localhost:8545",
+      chainId: 1101,
+      tags: ["local"]
+    },
+    mainnet: {
```

Comment:
We will define these data in an external file, please see https://github.com/keep-network/coverage-pools/commit/5e899ef7cc596f2bdf087f647676d7e8cdbccb46

Reply:
Ok, I will change. However, I'm curious about the motivations behind the approach you propose. I'm asking because creating a file from a template, filling it with data, and linking in the Hardhat config seems to be more cumbersome than just passing those things via env variables, as we used to do in other projects. Can you shed some light?

Reply:
See https://github.com/keep-network/tbtc-v2/pull/14/commits/6d74161552b3754d66ef2ba6d90ac3365addf7a0

Reply:
We used storing secret data in a file for our mainnet contracts migrations in the past. This protects better against secrets leaking during a screen share when you have to set env variables in a terminal when running a command.
It's also simpler to store the values in a file once and reuse it for other projects at different times. You can define the file in your home dir and point to this one file from all your hardhat configs across the projects.